types/src/front/api_handlers/public/data_sources.ts

```diff
@@ -35,6 +39,43 @@ export const PostDataSourceDocumentRequestBodySchema = t.type({
   async: t.union([t.boolean, t.undefined, t.null]),
 });
 
+export type PatchDataSourceDocumentRequestBody = t.TypeOf<
+  typeof PostDataSourceDocumentRequestBodySchema
+>;
+
+export const PostDataSourceDocumentRequestBodySchema = t.intersection([
+  t.type({
+    name: t.string,
+  }),
+  PatchDataSourceDocumentRequestBodySchema,
+]);
+
 export type PostDataSourceDocumentRequestBody = t.TypeOf<
   typeof PostDataSourceDocumentRequestBodySchema
 >;
+
+export const PatchDataSourceTableRequestBodySchema = t.type({
+  name: t.string,
+  description: t.string,
+  timestamp: t.union([t.number, t.undefined, t.null]),
+  tags: t.union([t.array(t.string), t.undefined, t.null]),
```

Comment:
It feels very weird to have all those nullable options 🤔

Reply:
Yes, very weird. I think we should keep only value or undefined, but seems decoder does not handles all cases properly, and we pass null in form body in some cases. Some cleanup needed here.